City-of-Helsinki/drupal-helfi-kymp
# UHF-9201 elasticsearh streets data

# [UHF-9201](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9201)

Fetch street data for street maintenance search.

## What was done
- Added new index for street data
- Added custom datasource which indexes data without using drupal entities
- Added custom street data datatype.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-9201_elastic_streets`
  * `make fresh`
* Run `make drush-cr`
* setup this one to prevent error on processors page https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/632

## How to test
- [Go to search api indexes
](https://helfi-kymp.docker.so/fi/kaupunkiymparisto-ja-liikenne/admin/config/search/search-api)
- See that street data index exists
- See that all street data index admin views works without problem
- Don't use cli to index just yet.
- Go to index view, Build tracking information for street data, rebuild should find a bit under 8000 hits
- Click the indexing button. all 8000 items should be indexed
- Do an elasticsearch query to see that the data exists (docker compose ps, check the elastic port)
 - localhost:{elastic port}/_search?size=1000 
 - Index should have data.
- Running drush commands for elastin indexing should work with no problems

 

## Other PRs
https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/632



[UHF-9201]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9201?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
